### PR TITLE
[test_nn] move dropout tests to test/nn/test_dropout.py

### PR DESCRIPTION
**Pull request by kshitij12345**
Ref https://github.com/pytorch/pytorch/issues/63085

### CODE REVIEW

File: test/nn/test_dropout.py
Comment (kshitij12345) [inline test/nn/test_dropout.py:186]: Somehow without this, the test fails.

File: test/nn/test_dropout.py
Comment (kshitij12345) [inline test/nn/test_dropout.py:184]: Are the tests under `test/nn/` run by xla?

Comment (kshitij12345) [reply test/nn/test_dropout.py:184]: Looking at the test logs seems like XLA is not running the tests under the newly added `test/nn/test*.py`. Should we open a new issue in xla?

cc: @JackCaoG

File: test/nn/test_dropout.py
Comment (kshitij12345) [reply test/nn/test_dropout.py:186]: Looks like the test is flaky : https://github.com/pytorch/pytorch/issues/84186

File: test/nn/test_dropout.py
Comment (JackCaoG) [reply test/nn/test_dropout.py:184]: oh yea, should be easy to fix. We just need to change [our test script in ](https://github.com/pytorch/xla/blob/master/test/run_tests.sh#L96).

Comment (JackCaoG) [reply test/nn/test_dropout.py:184]: @kshitij12345 This seems to be a simple pr on xla end, do you want to give it atry? Otherwise fell free to ping me again when this pr is approved so I will work on the xla fix.

**Review by albanD — APPROVED** [review]
SGTM!

File: test/nn/test_dropout.py
Comment (albanD) [reply test/nn/test_dropout.py:186]: could you tkae a look at why?

Comment (kshitij12345) [reply test/nn/test_dropout.py:186]: Sure.

File: test/nn/test_dropout.py
Comment (kshitij12345) [reply test/nn/test_dropout.py:184]: I will try a PR as there would be more files which would be split of `test_nn` and moved to `test/nn`.

File: test/nn/test_dropout.py
Comment (kshitij12345) [reply test/nn/test_dropout.py:186]: Flaky test failure is related to `dynamo` which I am not sure about.

```
2022-08-28T18:09:40.0858999Z   test_Dropout1d_cpu (__main__.TestNNDeviceTypeCPU) ... torchdynamo.convert_frame: [ERROR] WON'T CONVERT test_Dropout1d test_nn.py line 13722 
2022-08-28T18:09:40.0859583Z due to: 
2022-08-28T18:09:40.0859897Z Traceback (most recent call last):
2022-08-28T18:09:40.0860651Z   File "/var/lib/jenkins/.local/lib/python3.7/site-packages/torchdynamo/variables/functions.py", line 31, in wrap_bound_arg
2022-08-28T18:09:40.0861268Z     assert isinstance(val, VariableTracker), typestr(val)
2022-08-28T18:09:40.0862427Z AssertionError: memory_format
2022-08-28T18:09:40.0862666Z 
2022-08-28T18:09:40.0862790Z from user code:
2022-08-28T18:09:40.0863142Z    File "test_nn.py", line 13726, in test_Dropout1d
2022-08-28T18:09:40.0865885Z     self._test_dropout(nn.Dropout1d, device, input)
2022-08-28T18:09:40.0866139Z 
2022-08-28T18:09:40.0866370Z Set torchdynamo.config.verbose=True for more information
2022-08-28T18:09:40.0866728Z ==========
```

Comment (albanD) [reply test/nn/test_dropout.py:186]: cc @ezyang what should we do with these failures?

Comment (ezyang) [reply test/nn/test_dropout.py:186]: cc @anijain2305 

You can unblock by using `@skipIfTorchDynamo`

Comment (kshitij12345) [reply test/nn/test_dropout.py:186]: This PR is not blocked by the same.

**Pull request closed without merging**
